src/ui/CategoryFacet/CategoryValueChildrenRenderer.ts

```diff
@@ -0,0 +1,61 @@
+import { Dom } from '../../utils/Dom';
+import { CategoryFacetTemplates } from './CategoryFacetTemplates';
+import { CategoryValue, CategoryJsonValues, CategoryValueParent } from './CategoryValue';
+
+export class CategoryChildrenValueRenderer {
+  private children = [];
+  private listOfChildValues: Dom;
+
+  constructor(
+    private element: Dom,
+    private categoryFacetTemplates: CategoryFacetTemplates,
+    private categoryValueParent: CategoryValueParent
+  ) {}
+
+  public clearChildrenExceptOne(except: CategoryValue) {
+    const newChildren = [];
+    this.children.forEach(categoryValue => {
```

Comment:
Could use `.filter` here :

```
public clearChildrenExceptOne(except: CategoryValue) {
    this.children = this.children.filter(categoryValue => {
        if  (except === categoryValue) {
            return true;
        }
        categoryValue.clear();
        return false;
    });
}
```

Also possible to merge `clearChildren` and `clearChildrenExceptOne` by using the former function name and having `except` be an optional param. Will need to modify the `if` statement in this case to `if (except && except === categoryValue)`.